src/process.cr

```diff
@@ -415,6 +415,29 @@ class Process
   private def close_io(io)
     io.close if io
   end
+
+  # Changes the root directory and the current working directory for the current
+  # process.
+  #
+  # ```
+  # Process.chroot("/var/empty")
+  # ```
+  def self.chroot(path : String) : Nil
+    path.check_no_null_byte
+    return if (LibC.chroot(path) == 0 && LibC.chdir("/") == 0)
+
+    case (Errno.value)
```

Comment:
This case isn't required, `Errno.new` calls `strerror` and checks `Errno.value` for you:
```cr
if LibC.chroot(path) != 0
  raise Errno.new("Failed to chroot")
end

if LibC.chdir("/") != 0
  raise Errno.new("Failed to chdir after chroot")
end
```

would be the idiomatic way to write this method.

However, it does raise the issue of error handling, and that `Process.chroot` is now not atomic. If `chroot` succeeds but `chdir` fails, we're in some undefined state that the application is going to be *very* confused about.

Reply:
Corrected